modules/internal/NetworkAtomics.chpl

```diff
@@ -75,14 +76,30 @@ module NetworkAtomics {
       return ret:bool;
     }
 
+    inline proc compareExchange(ref expected:bool, desired:bool, param order: memoryOrder = memoryOrder.seqCst): bool {
+      pragma "insert line file info" extern externFunc("cmpxchg", int(64))
+        proc atomic_cmpxchg(ref expected:int(64), ref desired:int(64), l:int(32), obj:c_void_ptr, ref result:bool(32), succ:memory_order, fail:memory_order): void;
+
+      var ret:bool(32);
+      var te = expected:int(64);
+      var td = desired:int(64);
+      atomic_cmpxchg(te, td, _localeid(), _addr(), ret, c_memory_order(order), c_memory_order(readableOrder(order)));
+      expected = te:bool;
```

Comment:
And here also, though for a slightly different though related reason, namely that if the `atomic_cmpxchg()` does succeed, the value our caller passed for `expected` was already the "old value" which we want to return to them.